dsl/pace/dsl/stencil.py

```diff
@@ -290,13 +342,15 @@ def __init__(
             stencil_config: container for stencil configuration
             externals: compile-time external variables required by stencil
             skip_passes: compiler passes to skip when building stencil
+            timing_collector: Optional object that accumulates timings
         """
         if isinstance(origin, tuple):
             origin = cast_to_index3d(origin)
         origin = cast(Union[Index3D, Mapping[str, Tuple[int, ...]]], origin)
         self.origin = origin
         self.domain: Index3D = cast_to_index3d(domain)
         self.stencil_config: StencilConfig = stencil_config
+        self.timing_collector = timing_collector or TimingCollector()
```

Comment:
It's better to avoid relying on the implementation of `__nonzero__` for timing_collector by explicitly checking if it is None. Can you also make this a private attribute?